packages/website/private/contents/post-011.mdx

```diff
@@ -0,0 +1,99 @@
+---
+slug: rethinking-fine-grained-components
+title: Introducing “slice components”
+description: We’ve re-thought the api for fine-grained component rendering.
+author: tyler
+release: v0.24
+date: 2025/07/29
+---
+
+We're happy to announce Waku v0.24 with a brand new API: the Slice Components API! This will replace the page part component api, along with unlocking brand new functionality... `lazy` Slice Components.
+
+We heard the feedback around the Page Part components and feel like the Slices API will both solve a lot of the issues, while also unlocking brand new flexibility with render patterns!
+
+## Components are re-useable across pages
+
+One of the key issues with the Page Parts api is that each file is tied to a particular route. This is backwards from how we normally compose components. Usually the pattern is to write a component in some `_components` directory then import it to be used on each of your routes. Well... this is exactly how the slices api works!
+
+```
+ src/pages
+ ├── _components
+ │   ├── _slice-one.tsx
+ │   └── _slice-two.tsx
+ └── some-page.tsx
+```
+
+And some-page.tsx will be written something like this:
+
+```tsx
+import { Slice } from 'waku';
+
+export default function SomePage() {
+  return (
+    <div>
+      <Slice id="slice-one" />
+      <Slice id="slice-two" />
+    </div>
+  );
+}
+
+export const getConfig = () => {
+  return {
+    render: 'static',
+    slices: ['slice-one', 'slice-two'],
+  };
+};
+```
+
+Please notice: `slices` must be specified with each of the slice ids in order for waku to know how to bundle your page without any compiler magic.
+
+The file convention for Slice components is `_slice${string}.tsx` (TypeScript string literal syntax). We will look for these anywhere in the `src/pages` directory. And by default, the slice id will be the filename without the `_` prefix.
+
+An example for `_slice-one.tsx`:
+
+```tsx
+export default function SliceOne() {
+  return <p>🍕</p>;
+}
+
+export const getConfig = () => {
+  return {
+    id: 'slice-one',
```

Comment:
This can be omitted, right?

Reply:
it can be, but do we want the example to show how it's configurable, or no?

Reply:
> And by default, the slice id will be the filename without the `_` prefix.

As it's instructed, I thought we wanted to show the example. Or, comment it:

```suggestion
    id: 'slice-one', // can be omitted if it can be taken from the file name
```